docs/data/sdks/typescript-browser/migration.md

```diff
@@ -0,0 +1,497 @@
+  ```
+
+=== "@amplitude/analytics-browser"
+  ```typescript
+  const deviceId = "1"
+  amplitude.getDeviceId(deviceId)
+  ```
+
+#### `setSessionId()`
+
+Setting a session ID can be invoked on `amplitude` without calling `getInstance()`.
+
+=== "amplitude-js"
+  ```javascript
+  const sessionId = Date.now()
+  amplitude.getInstance().setSessionId(sessionId)
+  ```
+
+=== "@amplitude/analytics-browser"
+  ```typescript
+  const sessionId = Date.now()
+  amplitude.setSessionId(sessionId)
+  ```
+
+#### `clearUserProperties()`
+
+The `clearUserProperties` API has been removed; however the same funcationality can be achieved by using a unified `identify` API.
+
+=== "amplitude-js"
+  ```javascript
+  amplitude.getInstance().clearUserProperties()
+  ```
+
+=== "@amplitude/analytics-browser"
+  ```typescript
+  amplitude.identify(
+    new amplitude.Identify().identify.clearAll()
+  )
+  ```
+
+#### `setUserProperties()`
+
+The `clearUserProperties` API has been removed; however the same funcationality can be achieved by using a unified `identify` API.
+
+=== "amplitude-js"
+  ```javascript
+  amplitude.getInstance().setUserProperties({
+    membership, "paid",
+    payment, "bank",
+  })
+  ```
+
+=== "@amplitude/analytics-browser"
+  ```typescript
+  const identify = new amplitude.Identify()
+  identify
+    .set("membership", "paid")
+    .set("payment", "bank")
+  amplitude.identify(identify)
+  ```
+
+#### `identify()`
+
+Making an identify call can be done on `amplitude` without calling `getInstance()`.
+
+=== "amplitude-js"
+  ```javascript
+  const identify = new amplitude.Identify()
+  identify.set("membership", "paid")
+  amplitude.getInstance().identify(identify)
+  ```
+
+=== "@amplitude/analytics-browser"
+  ```typescript
+  const identify = new amplitude.Identify()
+  identify.set("membership", "paid")
+  amplitude.identify(identify)
+  ```
+
+### Setting group properties
+
+### `groupIdentify()`
+
+Making an identify call can be done on `amplitude` without calling `getInstance()`.
+
+=== "amplitude-js"
+  ```javascript
+  const identify = new amplitude.Identify()
+  identify.set("membership", "paid")
+  amplitude.getInstance().groupIdentify(identify)
+  ```
+
+=== "@amplitude/analytics-browser"
+  ```typescript
+  const identify = new amplitude.Identify()
+  identify.set("membership", "paid")
+  amplitude.groupIdentify(identify)
+  ```
+
+### Tracking revenue
+
+#### `logRevenueV2()`
+
+Tracking revenue can be done using `revenue()` API on `amplitude` without calling `getInstance()`.
+
+=== "amplitude-js"
+  ```javascript
+  const revenue = new amplitude.Revenue()
+  revenue
+    .setProductId("productId")
+    .setPrice(10)
+  amplitude.getInstance().logRevenueV2(revenue)
+  ```
+
+=== "@amplitude/analytics-browser"
+  ```typescript
+  const revenue = new amplitude.Revenue()
+  revenue
+    .setProductId("productId")
+    .setPrice(10)
+  amplitude.revenue(revenue)
+  ```
+
+### Patterns
+
+#### Plugins
```

Comment:
Do we need to mention how to add a plugin,  and also add the destination plugin example?

Reply:
Added